pkg/filters/builder/responseadaptor.go

```diff
@@ -31,51 +31,52 @@ import (
 )
 
 const (
-	// Kind is the kind of ResponseAdaptor.
-	Kind = "ResponseAdaptor"
+	// ResponseAdaptorKind is the kind of ResponseAdaptor.
+	ResponseAdaptorKind = "ResponseAdaptor"
 
 	resultResponseNotFound = "responseNotFound"
-	resultDecompressFailed = "decompressFailed"
-	resultCompressFailed   = "compressFailed"
-
-	keyContentLength   = "Content-Length"
-	keyContentEncoding = "Content-Encoding"
 )
 
-var kind = &filters.Kind{
-	Name:        Kind,
+var responseAdaptorKind = &filters.Kind{
+	Name:        ResponseAdaptorKind,
 	Description: "ResponseAdaptor adapts response.",
 	Results: []string{
 		resultResponseNotFound,
 		resultCompressFailed,
 		resultDecompressFailed,
 	},
 	DefaultSpec: func() filters.Spec {
-		return &Spec{}
+		return &ResponseAdaptorSpec{}
 	},
 	CreateInstance: func(spec filters.Spec) filters.Filter {
-		return &ResponseAdaptor{spec: spec.(*Spec)}
+		return &ResponseAdaptor{spec: spec.(*ResponseAdaptorSpec)}
 	},
 }
 
 func init() {
-	filters.Register(kind)
+	filters.Register(responseAdaptorKind)
 }
 
 type (
 	// ResponseAdaptor is filter ResponseAdaptor.
 	ResponseAdaptor struct {
-		spec *Spec
+		spec *ResponseAdaptorSpec
+		Builder
 	}
 
-	// Spec is HTTPAdaptor Spec.
-	Spec struct {
+	// ResponseAdaptorSpec is HTTPAdaptor ResponseAdaptorSpec.
+	ResponseAdaptorSpec struct {
 		filters.BaseSpec `json:",inline"`
+		Spec             `json:",inline"`
+
+		*ResponseAdaptorTemplate
```

Comment:
```suggestion
		*ResponseAdaptorTemplate `json:",inline"`
```

Reply:
Thanks for pointing out. As the validation goes, some fields of ResponseAdaptorTemplate should never be omitted, though some fields is empty or own a default value, but it not means the field could be omitted. If ResponseAdaptorTemplate is nil, it will be corruption. 

```golang
// Validate verifies that at least one of the validations is defined.
func (spec *RequestAdaptorSpec) Validate() error {
	if spec.Decompress != "" && spec.Decompress != "gzip" {
		return fmt.Errorf("RequestAdaptor only support decompress type of gzip")
	}
	if spec.Compress != "" && spec.Compress != "gzip" {
		return fmt.Errorf("RequestAdaptor only support decompress type of gzip")
	}
	if spec.Compress != "" && spec.Decompress != "" {
		return fmt.Errorf("RequestAdaptor can only do compress or decompress for given request body, not both")
	}
	if spec.Body != "" && spec.Decompress != "" {
		return fmt.Errorf("No need to decompress when body is specified in RequestAdaptor spec")
	}
	if spec.Sign == nil {
		return nil
	}
	s := spec.Sign
	if s.APIProvider != "" {
		if _, ok := signerConfigs[s.APIProvider]; !ok {
			return fmt.Errorf("%q is not a supported API provider", s.APIProvider)
		}
	}

	return nil
}
```